src/services/codefixes/convertToAsyncFunction.ts

```diff
@@ -0,0 +1,504 @@
+                const symbolIdString = getSymbolId(symbol).toString();
+
+                // if the identifier refers to a function we want to add the new synthesized variable for the declaration (ex. blob in let blob = res(arg))
+                // Note - the choice of the first call signature is arbitrary 
+                if (callSignatures && callSignatures.length > 0 && callSignatures[0].parameters.length && !synthNamesMap.has(symbolIdString)) {
+                    const synthName = getNewNameIfConflict(createIdentifier(callSignatures[0].parameters[0].name), allVarNames);
+                    synthNamesMap.set(symbolIdString, synthName);
+                    allVarNames.push({ identifier: synthName.identifier, symbol });
+                }
+                // we only care about identifiers that are parameters and declarations (don't care about other uses)
+                else if (node.parent && (isParameter(node.parent) || isVariableDeclaration(node.parent))) {
+
+                    // if the identifier name conflicts with a different identifier that we've already seen
+                    if (allVarNames.some(ident => ident.identifier.text === node.text && ident.symbol !== symbol)) {
+                        const newName = getNewNameIfConflict(node, allVarNames);
+                        identsToRenameMap.set(symbolIdString, newName.identifier);
+                        synthNamesMap.set(symbolIdString, newName);
+                        allVarNames.push({ identifier: newName.identifier, symbol });
+                    }
+                    else {
+                        const identifier = getSynthesizedDeepClone(node);
+                        identsToRenameMap.set(symbolIdString, identifier);
+                        synthNamesMap.set(symbolIdString, { identifier, numberOfAssignmentsOriginal: allVarNames.filter(elem => elem.identifier.text === node.text).length, numberOfAssignmentsSynthesized: 0 });
+                        if ((isParameter(node.parent) && isCallbackOnTypePromise(node.parent.parent, setOfAllCallbacksToReturn)) || isVariableDeclaration(node.parent)) {
+                            allVarNames.push({ identifier, symbol });
+                        }
+                    }
+                }
+            }
+        });
+
+        function isCallbackOnTypePromise(child: Node, setOfAllCallbacksToReturn: Map<true>): boolean {
+            let node = child.parent;
+            if (isCallExpression(node) || isIdentifier(node) && !setOfAllCallbacksToReturn.get(getNodeId(node).toString())) {
+                const nodeType = checker.getTypeAtLocation(node);
+                const isPromise = nodeType && checker.getPromisedTypeOfPromise(nodeType);
+                return !!isPromise;
+            }
+
+            return false;
+        }
+
+        function deepCloneCallback(node: Node, clone: Node) {
+            const type = checker.getTypeAtLocation(node);
+            const symbol = checker.getSymbolAtLocation(node);

```

Comment:
I'm pretty nervous about the perf implications of calling `getSymbolAtLocation` on *every* cloned node.  At minimum, I'd suggest checking whether `node` is an identifier first.  It might also be worthwhile to add a layer of indirection to `identsToRenameMap` - name -> symbol ID -> new name.